### PR TITLE
Ref/motor logging

### PR DESCRIPTION
**Pull request by joshLong145**
addition of `motor/x/logger` for logging within `pkg/motor`

## Changes

* refactored all `fmt` print statements to use the logger now within `motorNodeImpl`
* added `pkg/motor/x/logger` for all internal logging within the motor package
* logger has levels and ability to format strings.

## Checklist

* [ ] Unit tests
* [ ] Documentation

## References *(optional)*

Fixes
Connects

 Mention [stepsize] in a comment if you'd like to report some technical debt. See examples here.

### CODE REVIEW

File: pkg/motor/x/logger/logger.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+		level:    conevertLogLevel(level),
+		category: category,
+		logger:   log.New(os.Stdout, category, 0),
+	}
+}
+
+func (l *Logger) log(level LogLevel, category string, message string, args ...interface{}) {
+	if l.level <= level {
+		ts := time.Now().Format("2006-01-02 15:04:05")
+		prefix := fmt.Sprintf("%s %s %s : ", conevertLogLevelCategory(level), ts, l.category)
```
Comment (mcjcloud) [inline pkg/motor/x/logger/logger.go]: You accept `category` as a parameter but don't use it.

File: pkg/motor/x/logger/logger.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	CAT_FATAL = "[FATAL]"
+)
+
+type Logger struct {
+	logger   *log.Logger
+	level    LogLevel
+	category string
+}
+
+func New(level string, category string) *Logger {
```
Comment (mcjcloud) [inline pkg/motor/x/logger/logger.go]: It looks like the `category` isn't actually used anywhere. Does it need to be part of the struct?

File: pkg/motor/x/logger/utils.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
+package logger
+
+func conevertLogLevelCategory(level LogLevel) string {
+	switch level {
+	case LEVEL_INFO:
+		return "[INFO]"
```
Comment (mcjcloud) [inline pkg/motor/x/logger/utils.go]: nit: Use the constants here

File: pkg/motor/x/logger/utils.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+	case LEVEL_ERROR:
+		return "[ERROR]"
+	case LEVEL_FATAL:
+		return "[FATAL]"
+	}
+
+	return ""
+}
+
+func conevertLogLevel(level string) LogLevel {
```
Comment (mcjcloud) [inline pkg/motor/x/logger/utils.go]: Where does the string version of these levels come from?

File: pkg/motor/motor.go [hunk 5]
```diff
@@ -67,10 +77,12 @@ func EmptyMotor(r *mt.InitializeRequest, cb common.MotorCallback) (*motorNodeImp
 		supportDir:         r.GetSupportDir(),
 		tempDir:            r.GetTempDir(),
 		callback:           cb,
+		logLevel:           r.LogLevel,
 	}, nil
 }
 
 func initMotor(mtr *motorNodeImpl, options ...mpc.WalletOption) (err error) {
+	mtr.log = logger.New(mtr.logLevel, "motor")
```
Comment (mcjcloud) [inline pkg/motor/motor.go:84]: I'm confused as to what the second argument here is for.

**Review by mcjcloud — CHANGES_REQUESTED** [review]
I think this package would be useful outside of motor, like directly in `pkg/`

File: pkg/motor/x/logger/utils.go (hunk 4 above)
Comment (joshLong145) [reply pkg/motor/x/logger/utils.go]: it comes from motor as to allow for strings in the request for initialize

**Comment by joshLong145** [comment]
> I think this package would be useful outside of motor, like directly in `pkg/`

has been moved to 'pkg'

**Review by prdlk — APPROVED** [review]

**Comment by prdlk** [comment]
> I think this package would be useful outside of motor, like directly in `pkg/`

Maybe internal?

File: proto/motor/v1/request.proto [hunk 6]
```diff
@@ -168,3 +170,19 @@ message SeachBucketContentBySchemaRequest {
 }
 
 message StatRequest {}
+
+message UploadDocumentRequest {
```
Comment (mcjcloud) [inline proto/motor/v1/request.proto:174]: Why is this in this PR?

Comment (joshLong145) [reply proto/motor/v1/request.proto:174]: think it was from a merger i did and committed by mistake, good catch. removing this

**Review by mcjcloud — APPROVED** [review]

**Pull request merged**